config/confighttp/confighttp.go

```diff
@@ -74,6 +74,10 @@ type HTTPClientSettings struct {
 	// IdleConnTimeout is the maximum amount of time a connection will remain open before closing itself.
 	// There's an already set value, and we want to override it only if an explicit value provided
 	IdleConnTimeout *time.Duration `mapstructure:"idle_conn_timeout"`
+
+	// DisableKeepAlives, if true, disables HTTP keep-alives and will only use the connection to the server
+	// for a single HTTP request.
```

Comment:
```suggestion
	// for a single HTTP request.
	//
	// WARNING: enabling this option can result in significant overhead establishing a new HTTP(S)
	// connection for every request.  Before enabling this option please consider whether changes
	// to idle connection settings can achieve your goal.
```

Reply:
@Aneurysm9 I added this warning

Reply:
I wonder if this should be a warning for the user of the collector as well, visible in the collector's logs when this option is enabled.

Reply:
I feel like this warning, if it was visible in the collector logs, would potentially just be noise to someone who's already aware of the configuration they're using and need to have this option disabled.